metricbeat/module/mongodb/replstatus/info.go

```diff
@@ -43,85 +43,112 @@ type CollSize struct {
 	}
 
-	collection := db.Collection(oplogCol)
-
-	// get oplog size
+	// Get MaxSize and Size from collStats by using db.runCommand
 	var oplogSize CollSize
-	res := db.RunCommand(context.Background(), bson.D{bson.E{Key: "collStats", Value: oplogCol}})
-	if err = res.Err(); err != nil {
-		return nil, fmt.Errorf("'collStats' command returned an error: %w", err)
+	if err := res.Decode(&oplogSize); err != nil {
+		return nil, fmt.Errorf("could not decode mongodb oplog size: %w", err)
 	}
 
-	if err = res.Decode(&oplogSize); err != nil {
-		return nil, fmt.Errorf("could not decode mongodb op log size: %w", err)
-	}
-
-	// get first and last items in the oplog
+	collection := db.Collection(oplogCol)
 	firstTs, lastTs, err := getOpTimestamp(collection)
 	if err != nil {
-		return nil, fmt.Errorf("could not get operation timestamp in op log: %w", err)
+		return nil, fmt.Errorf("could not get operation timestamps from oplog: %w", err)
 	}
 
-	diff := lastTs - firstTs
-
-	return &oplogInfo{
+	info := &oplogInfo{
 		allocated: oplogSize.MaxSize,
 		used:      oplogSize.Size,
 		firstTs:   firstTs,
 		lastTs:    lastTs,
-		diff:      diff,
-	}, nil
-}
+		diff:      lastTs - firstTs,
+	}
 
-func getOpTimestamp(collection *mongo.Collection) (uint32, uint32, error) {
+	return info, nil
+}
 
-	// Find both first and last timestamps using $min and $max
-	pipeline := bson.A{
-		bson.M{"$group": bson.M{"_id": 1, "minTS": bson.M{"$min": "$ts"}, "maxTS": bson.M{"$max": "$ts"}}},
+// getOpTimestamp returns the first and last timestamp of the oplog collection.
+func getOpTimestamp(collection *mongo.Collection) (int64, int64, error) {
+	// NOTE(shmsr):
+	//
+	// When you do db.getReplicationInfo() in monogo shell (mongosh), you can see
+	// 	{
+	//		...
+	// 		tFirst: 'Tue Jan 07 2025 22:33:28 GMT+0530 (India Standard Time)',
+	// 		tLast: 'Wed Jan 08 2025 11:45:07 GMT+0530 (India Standard Time)',
+	// 		now: 'Wed Jan 08 2025 11:45:14 GMT+0530 (India Standard Time)'
+	// 	}
+	// i.e., we get tFirst and tLast from oplog.rs which is the first and last
+	// timestamp of the oplog.
+	// Source from the same is here:
+	// 	https://github.com/mongodb/mongo/blob/20cbee37a0ee4d40b35d08b6a34ade81252f86a8/src/mongo/shell/db.js#L863
+	// This is how they calculate tFirst and tLast:
+	// 	https://github.com/mongodb/mongo/blob/20cbee37a0ee4d40b35d08b6a34ade81252f86a8/src/mongo/shell/db.js#L889
+	// So ideally, we will replicate the same logic here:
+	// 	var firstc = ol.find().sort({$natural: 1}).limit(1);
+	// 	var lastc = ol.find().sort({$natural: -1}).limit(1);
+	//
+	// oplog.rs is designed to scanned in natural ($natural) order. So, when
+	// querying without any sort, it will return the first entry in natural order.
+	// When we sort in reverse natural order (i.e., $natural: -1), it will return
+	// the last entry in natural order.
+	//
+	// The queries use optimizations like FindOne() and SetProjection() to limit
```

Comment:
Do we need to put this information in the code. (About prohection and findone functions ? Looks TMI ?)
We can add this to the PR description or the isue. If someone wants details they can refer there.

Reply:
Ok fine I'll remove it. Anyway the methods are documented, so one can always refer mongodb's driver doc.